src/main/kotlin/com/ampnet/userservice/controller/CoopController.kt

```diff
@@ -16,14 +17,19 @@ import org.springframework.web.bind.annotation.RestController
 import javax.validation.Valid
 
 @RestController
-class CoopController(private val coopService: CoopService) {
-
+class CoopController(
+    private val coopService: CoopService,
+    private val reCaptchaService: ReCaptchaService
+) {
     companion object : KLogging()
 
     @PostMapping("/coop")
     @CacheEvict(value = [COOP_CACHE], allEntries = true)
     fun createCoop(@Valid @RequestBody request: CoopRequest): ResponseEntity<CoopServiceResponse> {
         logger.info { "Received request to create coop: $request" }
+        request.reCaptchaToken?.let {
+            reCaptchaService.validateResponseToken(it)
+        }
```

Comment:
This logic is wrong. If the user doesn't send the token it will never validate token even if set in the config that Recaptcha is needed. Pass a nullable string to `reCaptchaService` and validate token depending on the config.